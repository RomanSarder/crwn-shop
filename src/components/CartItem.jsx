import React from 'react'
import styled from 'styled-components'

var StyledCartItem = styled.div`
    width: 100%;
    display: flex;
    min-height: 10rem;
    margin-bottom: 1.5rem;

    img {
        width: 25%;
        height: 95%;
        object-fit: cover;
    }

    .details {
        width: 70%;
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        justify-content: center;
        padding: 1rem 2rem;
    }

    .subtotal {
        display: flex;
        align-items: center;
        
        span {
            font-size: 2.6rem;
            font-weight: bold;
        }
    }
`

export default function CartItem( { item: { id, price, imageUrl, name, quantity } } ) {
    return (
        <StyledCartItem>
            <img src={imageUrl} alt={name} />
            <div className="details">
                <span>{name}</span>
                <span>{quantity} * ${price}</span>
            </div>
            <div className="subtotal">
                <span>${quantity * price}</span>
            </div>
        </StyledCartItem>
    )
}
