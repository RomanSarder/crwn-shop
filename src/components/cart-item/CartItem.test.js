import { render } from '@testing-library/react'
import 'jest-styled-components'
import CartItem from './CartItem'

it('<CartItem /> should render cart item properly', () => {
    var testItem = { price: 10, imageUrl: 'test-url', name: 'test item', quantity: 2 }
    var { container } = render(<CartItem item={testItem}/>)

    expect(container.firstChild).toMatchSnapshot()
})