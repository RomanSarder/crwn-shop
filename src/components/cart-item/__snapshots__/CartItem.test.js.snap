// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should render cart item properly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 10rem;
  margin-bottom: 1.5rem;
}

.c0 img {
  width: 25%;
  height: 95%;
  object-fit: cover;
}

.c0 .details {
  width: 70%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 2rem;
}

.c0 .subtotal {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .subtotal span {
  font-size: 2.6rem;
  font-weight: bold;
}

<div
  class="c0"
>
  <img
    alt="test item"
    src="test-url"
  />
  <div
    class="details"
  >
    <span>
      test item
    </span>
    <span>
      2
       * $
      10
    </span>
  </div>
  <div
    class="subtotal"
  >
    <span>
      $
      20
    </span>
  </div>
</div>
`;
